### ISSUE TITLE
【ERROR】Mongo ACL Error: something wrong with mongo replica in ACL

### ISSUE THREAD

**Issue by phper-chen** [emqx/emqx#2187]
#### Environment

- OS: unbuntu
- Erlang/OTP: 
- EMQ: 3.0 release

#### Description

*A description of the issue*
I have already open the mongo acl plugin, but it always closes automicly, so I have to open it mannualy. what I found on the log below:
![image](https://user-images.githubusercontent.com/20152358/51814977-aec95100-2283-11e9-9ee6-f268a16925c3.png)

**Comment by phper-chen** [comment]
The config info below:
![image](https://user-images.githubusercontent.com/20152358/51815107-737b5200-2284-11e9-9dc7-42b82213fe29.png)

**Comment by phper-chen** [comment]
![image](https://user-images.githubusercontent.com/20152358/51815312-a70aac00-2285-11e9-9979-d370b9564288.png)

**Comment by linjunjj** [comment]
Hi @phper-chen  It seems that your `auth. mongo. server` configuration is wrong

**Comment by phper-chen** [comment]
> Hi @phper-chen It seems that your `auth. mongo. server` configuration is wrong

wut should I do? should I add a config "auth.mongo.type=rs" in mongo replica? Thank u!

**Comment by linjunjj** [comment]
@phper-cher yeah. Does your mongoDB use Replica Set ？

**Comment by phper-chen** [comment]
> @phper-chen yeah. Does your mongoDB use Replica Set ？

yes, I used a url like this "mongodb://dds-xxxxxxxxx.mongodb.rds.aliyuncs.com:3909/example?replicaSet=mgset-201893"
In log, it shows below

**Comment by linjunjj** [comment]
> > @phper-chen yeah. Does your mongoDB use Replica Set ？
> 
> yes, I used a url like this "mongodb://dds-xxxxxxxxx.mongodb.rds.aliyuncs.com:3909/example?replicaSet=mgset-201893"
> In log, it shows below, and this error leads a high cpu usage.
> ![image](https://user-images.githubusercontent.com/20152358/51820917-e1cc0e80-229c-11e9-9bfe-e4aec9969738.png)

@phper-chen  The `auth. mongo. server` configuration only supports `host + port` format. You can try this `dds-xxxxxxxxx.mongodb.rds.aliyuncs.com:3909`

**Comment by phper-chen** [comment]
My env in docker-compose:
![image](https://user-images.githubusercontent.com/20152358/51836745-e3123100-22c6-11e9-9a97-7ec2232894ee.png)

And it always show these error:
![image](https://user-images.githubusercontent.com/20152358/51836814-22408200-22c7-11e9-8065-d95cdc288a6d.png)

**Comment by linjunjj** [comment]
> My env in docker-compose:
> ![image](https://user-images.githubusercontent.com/20152358/51836745-e3123100-22c6-11e9-9a97-7ec2232894ee.png)
> 
> And it always show these error:
> ![image](https://user-images.githubusercontent.com/20152358/51836814-22408200-22c7-11e9-8065-d95cdc288a6d.png)

It seems that password error. Is your MongoDB account password and url correct?

**Comment by phper-chen** [comment]
> > My env in docker-compose:
> > ![image](https://user-images.githubusercontent.com/20152358/51836745-e3123100-22c6-11e9-9a97-7ec2232894ee.png)
> > And it always show these error:
> > ![image](https://user-images.githubusercontent.com/20152358/51836814-22408200-22c7-11e9-8065-d95cdc288a6d.png)
> 
> It seems that password error. Is your MongoDB account password and url correct?

Maybe I should set the encryption method of the mongo replica set connection password to “SCRAM-SHA-1”，Thank you for your patience，and I will have a try~
